File: t/CS/RasterizeDocumentDemo/LoadPagesDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace RasterizeDocumentDemo
{
   public partial class LoadPagesDialog : Form
   {
      public LoadPagesDialog()
      {
         InitializeComponent();
      }

      private int _totalPages;

      [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
      public int TotalPages
      {
         get { return _totalPages; }
         set { _totalPages = value; }
      }

      private int _firstPageNumber;

      [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
      public int FirstPageNumber
      {
         get { return _firstPageNumber; }
         set { _firstPageNumber = value; }
      }

      private int _lastPageNumber;

      [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
      public int LastPageNumber
      {
         get { return _lastPageNumber; }
         set { _lastPageNumber = value; }
      }

      protected override void OnLoad(EventArgs e)
      {
         if(!DesignMode)
         {
            _documentPagesControl.TotalPages = _totalPages;
            _documentPagesControl.FirstPageNumber = _firstPageNumber;
            _documentPagesControl.LastPageNumber = _lastPageNumber;
            _documentPagesControl.SetData();
         }

         base.OnLoad(e);
      }

      private void _okButton_Click(object sender, EventArgs e)
      {
         if(_documentPagesControl.GetData())
         {
            _totalPages = _documentPagesControl.TotalPages;
            _firstPageNumber = _documentPagesControl.FirstPageNumber;
            _lastPageNumber = _documentPagesControl.LastPageNumber;
         }
         else
         {
            DialogResult = DialogResult.None;
         }
      }
   }
}

File: t/VB/RasterizeDocumentDemo/LoadPagesDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System.ComponentModel

Public Class LoadPagesDialog
   Private _totalPages As Integer

   <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
   Public Property TotalPages() As Integer
      Get
         Return _totalPages
      End Get
      Set(ByVal value As Integer)
         _totalPages = value
      End Set
   End Property

   Private _firstPageNumber As Integer

   <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
   Public Property FirstPageNumber() As Integer
      Get
         Return _firstPageNumber
      End Get
      Set(ByVal value As Integer)
         _firstPageNumber = value
      End Set
   End Property

   Private _lastPageNumber As Integer

   <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
   Public Property LastPageNumber() As Integer
      Get
         Return _lastPageNumber
      End Get
      Set(ByVal value As Integer)
         _lastPageNumber = value
      End Set
   End Property

   Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)
      If Not DesignMode Then
         _documentPagesControl.TotalPages = _totalPages
         _documentPagesControl.FirstPageNumber = _firstPageNumber
         _documentPagesControl.LastPageNumber = _lastPageNumber
         _documentPagesControl.SetData()
      End If

      MyBase.OnLoad(e)
   End Sub

   Private Sub _okButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _okButton.Click
      If _documentPagesControl.GetData() Then
         _totalPages = _documentPagesControl.TotalPages
         _firstPageNumber = _documentPagesControl.FirstPageNumber
         _lastPageNumber = _documentPagesControl.LastPageNumber
      Else
         DialogResult = DialogResult.None
      End If
   End Sub
End Class
